public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/reducer.ts

```diff
@@ -78,10 +78,13 @@ export const queriesAndExpressionsReducer = createReducer(initialState, (builder
     })
     .addCase(updateExpression, (state, { payload }) => {
       state.queries = state.queries.map((query) => {
+        const dataSource = state.queries.find((alertQuery) => alertQuery.refId === payload.expression);
```

Comment:
Could we add a reducer's test for this case?

Reply:
yes!